daemon/access.go

```diff
@@ -198,6 +198,7 @@ func requireInterfaceApiAccessImpl(d *Daemon, ucred *ucrednet, interfaceName str
 			return Forbidden("internal error: %s", err)
 		}
 		if connRef.PlugRef.Snap == snapName {
+			r.RemoteAddr = ucrednetAttachInterface(r.RemoteAddr, interfaceName)
```

Comment:
```suggestion
			// since a connection associated with ucred may be reused and so  live longer
			//  than the interface connection, only update the data associated with the immediate
			// request
			r.RemoteAddr = ucrednetAttachInterface(r.RemoteAddr, interfaceName)
```

Reply:
bit crazy, but we could abuse request context and add a way to adding some sort of security/interfaces data